packages/shopify/react/components/LiquidBlock.tsx

```diff
@@ -37,19 +38,24 @@ export const LiquidBlock = ({
       if (cache[cacheKey]) {
         return;
       }
+
       // Convert `sections/foo.liquid` -> `foo`
       if (!blockName) {
         return;
       }
+
       if (cache[cacheKey]) {
         setHtml(cache[cacheKey]);
         return;
       }
 
+      const previewThemeID = builderState?.state?.location?.query?.preview_theme_id;
```

Comment:
did you test this and confirm this is working?

iirc shopify removes the preview_theme_id param automatically (301) and so for other stuff I had to pass a new theme_id param in the editor and use that instead

we may want to also check the current URL theme ID since this stuff really only applies client side anyway and iirc state.location.query can be spotty in some contexts

```suggestion
      const previewThemeID = builderState?.state?.location?.query?.theme_id || (Builder.isBrowser && getQueryParam('theme_id');
```

Reply:
actually I think there is something better - on any shoipfy store I'm seeing a window variable for this

`Shopify.theme.id` - maybe we use that

```suggestion
      const previewThemeID = builderState?.state?.location?.query?.preview_theme_id || (Builder.isBrowser && window.Shopify?.theme?.id;
```

Reply:
let's prioritize getting this in and then we can do a webcomponents dev release

Reply:
Yeah, I got the `builderState?.state?.location?.query?.preview_theme_id` from the debugger, and so far is always there.

And you are right, `window.Shopify?.theme?.id` is also present so far, and its value also match our `query.preview_theme_id` so far.

Actually I kind of prefer going with your suggestion, but swapping preferences, like:
```suggestion
      const previewThemeID = Builder.isBrowser && window.Shopify?.theme?.id || builderState?.state?.location?.query?.preview_theme_id;
```
Thoughts?

Reply:
👍